rest/route/host_create.go

```diff
@@ -70,7 +70,40 @@ func (h *hostCreateHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 
 	ids := []string{}
+	t, err := task.FindOneId(h.taskID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding task '%s'", h.taskID))
+	}
+	if t == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
+		})
+	}
+	initialHosts, err := host.FindHostsSpawnedByTask(ctx, h.taskID, t.Execution)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding hosts spawned by task '%s'", h.taskID))
+	}
+	numInitialHosts := len(initialHosts)
 	for i := 0; i < numHosts; i++ {
+		// If there are an unexpected number of host intents, log a message and continue.
```

Comment:
nit: move this into the `if` block since it's for special case checking and doesn't actually apply to the logic after L105.

I also think we don't need to restate that the code logs a message and continues (since you could see  that's what it does on L96-104). It's more helpful to explain why it has to do this check (i.e. if the request is retried, it should still make the correct total number of hosts).